src/data/usage-data-service.js

```diff
@@ -4,17 +4,35 @@ const baseUrl = process.env.apiUrl
 
 // Analytics API is available at `http://api.openverse.engineering/analytics/`
 // and not `http://api.openverse.engineering/v1/analytics
-const ApiService = createApiService(baseUrl.replace('/v1', ''))
+const ApiService = createApiService(baseUrl?.replace('/v1', ''))
+
+/** @typedef {'search_event' | 'result_click_event' | 'detail_page_event' | 'search_rating_event'} SearchEventName */
 
 const UsageDataService = {
-  post(endpoint, params) {
-    return ApiService.post(`analytics/${endpoint}`, params)
+  /**
+   * @param {SearchEventName} endpoint
+   * @param {Parameters<typeof ApiService['post']>[1]} data
+   */
+  post(endpoint, data) {
+    return ApiService.post(`analytics/${endpoint}`, data)
   },
 
+  /**
+   * @param {object} props
+   * @param {string} props.query
+   * @param {string} props.sessionId
+   */
   sendSearchQueryEvent({ query, sessionId }) {
     return this.post('search_event', { query, session_uuid: sessionId })
   },
 
+  /**
+   * @param {object} props
+   * @param {string} props.query
+   * @param {unknown} props.resultRank
```

Comment:
This type can be set to a number because it's a result of a `findIndex` call: https://cs.github.com/WordPress/openverse-frontend/blob/af82bcd215933032fd3c557495bd32d41866b44d/src/store/media.js#L170